### ISSUE TITLE
Text tree view of nodes

### ISSUE THREAD

**Issue by WillSmithTE** [WillSmithTE/ameliorate#111]
### Describe your issue

If I want to scan through all nodes of a type, using the node types filter and looking through the diagram is ok, but seeing relations in this case is not necessary, and so the spread-out nodes doesn't provide extra value and is just distracting. It'd be nice to have a view that's more organized and compressed for reading through a bunch of nodes.

This is particularly nice if I just want to see all the questions, or all the sources and facts, or all the arguments that exist.

### Solution you'd like

A tree view seems like it'd be good for this (example from vscode):
![image](https://github.com/amelioro/ameliorate/assets/13872370/de2b61a6-b0fc-45c7-9f3d-7d8ed7824697)

- "Tree" format (vs diagram/table)
- Having a tree view branch for the child relations of the node type could be a really convenient way to quickly add facts for sources, but this should probably be done separately to keep scope of this ticket narrow

#### Example situations
- for [organizing-Ameliorate-diagrams](https://ameliorate.app/keyserj/organizing-Ameliorate-diagrams?view=Modeling+intuitions+-+scores), it'd be really nice to show the solutions at the root, and be able to expand each once to see their neighbors (components, then nested components/effects)

### Questions to answer

- Should this tree be viewable in the same view as the diagram/table, or should it always be in the side pane?
  - seems like usually this is ideal in the side pane, but _could_ have value as the main view? similar to the summary view...
- Should the tree view be filtered the same as the diagram is? That seems good and basically free to implement
  - maybe this would be good as long as we can choose specifically which nodes should be at the root of the tree
  - hmm there might be more benefit to being a little looser on filtering though, e.g. easily being able to show questions/facts in the tree view (kind of annoying to do this in the diagram right now?)
- Can this be built via yml/JSON? Maybe this could be a human-readable way to edit a diagram? Maybe hard though because editing would require node id's in the yml?
  - could build the yml/JSON with id's to be human editable, and just have the tree not show id's
- Could the tree view default to parent->child direction, but flip the other way via toggle? e.g. questions might be better parent->child, but sources seem sometimes better child->parent for adding facts
  - note: probably "incoming"/"outgoing" is less-ambiguous terminology than child/parent
  - might be better to expand to both parent and child nodes, could distinguish via a symbol
    - would this be doable even if expanding parents/children could lead to infinitely-nested expansion?

### Technical ideas

- maybe [MUI tree view](https://mui.com/x/react-tree-view/), but it looks a bit new and have some features that are paid-only
- [arborist](https://github.com/brimdata/react-arborist) also looks like a relatively popular tree view component for react
